django/contrib/gis/gdal/raster/source.py

```diff
@@ -472,3 +475,6 @@ def info(self):
         if not capi.get_ds_info:
             raise ValueError('GDAL ≥ 2.1 is required for using the info property.')
         return capi.get_ds_info(self.ptr, None).decode()
+
+    def clone(self):
+        return GDALRaster(capi.copy_ds(self.ptr))
```

Comment:
What about

```suggestion
        return type(self)(capi.copy_ds(self.ptr))
```

Should `self._write` also be passed?